### PR TITLE
Convert lib/sources to TypeScript

### PR DESCRIPTION
**Pull request by matsjla**
Backend code to TypeScript yay

### CODE REVIEW

File: lib/sources/builtin.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2022, Compiler Explorer Authors
```
Comment (RubenRBS) [inline lib/sources/builtin.ts:1]: Pitty git does not see this one as a rename too

File: lib/sources/builtin.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import fs from 'fs';
+import fsp from 'fs/promises';
+import path from 'path';
+
+import * as props from '../properties';
+
+import type { SourceEntry, Source } from './index';
```
Comment (RubenRBS) [inline lib/sources/builtin.ts:31]: This linter checks should be fixed before merging

**Review by RubenRBS — APPROVED** [review]
Uhh! Nice! The more ts, the better. Is it simple to expand the prettier config once that pr is merged to also handle lib code?

**Comment by matsjla** [comment]
> Uhh! Nice! The more ts, the better. Is it simple to expand the prettier config once that pr is merged to also handle lib code?

Yeah, can just add lib/*.ts to the package-json glob

**Pull request merged**